docs/src/solvers/ode_solve.md

```diff
@@ -493,6 +493,11 @@ These methods also have option `nlsolve` same as SDIRK methods. These methods al
 - `GRK4A` - An A-stable 4th order Rosenbrock method. Essentially "anti-L-stable"
   but efficient.
 - `Ros4LStab` - A 4th order L-stable Rosenbrock method.
+- `RosenbrockW6S4OS` - A 4th order L-stable Rosenbrock-W method (fixed step only).
+- `ROS34PW1a` - A 4th order L-stable Rosenbrock-W method.
```

Comment:
These support mass matrices too, right? That should get mentioned in the DAE page (and a test)

Reply:
Yes, but out-of-place methods don't support it, just like others.

Reply:
Add it to the DAE page, and we'll fix up the out of place ones. I am hoping this gets done this week BTW, either me, you, @YingboMa , or @ranjanan should probably take a look at it.

Reply:
I can fix the generic one.

Reply:
IIRC what needs to be fixed is the oop W generation and that's it? So if you fix one you fix all.

Reply:
Not really, the oop stepping code ignored mul of mass_matrix on du.
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/abaf387213e1b1b020b795242cf4bd5ca2a950d8/src/perform_step/rosenbrock_perform_step.jl#L129-L135
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/abaf387213e1b1b020b795242cf4bd5ca2a950d8/src/perform_step/rosenbrock_perform_step.jl#L217

Reply:
that doesn't need to be modified since it's in the W matrix.  W = M - gamma*J instead of W = I - gamma*J to generalize it, and it uses the same linear solve.

Reply:
But why inplace code have that?

Reply:
Oh right, I forgot that part lol